src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -486,14 +486,20 @@ private void ApplyDecompressionSetting(HttpRequestMessage request, Decompression
             {
```

Comment:
Should the ClientCertificateOption.Automatic case (lines 508 through 517 below) leverage the _CertificateHelper.GetEligibleClientCertificate_ variant that doesn't take an input collection? Right now, the certs are being picked solely by the ClientAuthenticationOID criteria, but CertificateHelper.GetEligibleClientCertificate has additional filters.